### PR TITLE
Fix #13895 - Support TypeVar in quotes printers

### LINKED ISSUE #13895: Quoted Expr contains unexpected TypeVar
## Compiler version

3.1.2-RC1-bin-20211022-f7abd32-NIGHTLY

## Minimized code

TODO

Quite hard to draft a reproducer as would depend on internal types not exposed in the API (`TypeVar`, `TypeParamRef`, ...) : https://github.com/cchantep/ReactiveMongo-BSON/blob/task/dotty/api/src/main/scala-3/MacroImpl.scala#L338

## Output

```
[error]    |scala.MatchError: TypeVar(TypeParamRef(Opts)) (of class dotty.tools.dotc.core.Types$TypeVar)
[error]    |	at scala.quoted.runtime.impl.printers.Extractors$ExtractorsPrinter.visitType(Extractors.scala:242)
[error]    |	at scala.quoted.runtime.impl.printers.Extractors$.showType(Extractors.scala:12)
[error]    |	at scala.quoted.runtime.impl.QuotesImpl$$anon$17.show(QuotesImpl.scala:2960)
[error]    |	at scala.quoted.runtime.impl.QuotesImpl$$anon$17.show(QuotesImpl.scala:2959)
[error]    |	at scala.quoted.runtime.impl.QuotesImpl$reflect$TypeReprMethods$.show(QuotesImpl.scala:1708)
[error]    |	at scala.quoted.runtime.impl.QuotesImpl$reflect$TypeReprMethods$.show(QuotesImpl.scala:1708)
```

## Expectation

Print successfully the `Expr`

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: compiler/src/scala/quoted/runtime/impl/printers/SourceCode.scala [hunk 1]
```diff
@@ -1234,6 +1236,9 @@ object SourceCode {
         this += " => "
         printType(rhs)
 
+      case tv: core.Types.TypeVar =>
+        this += tv.toString()
```
Comment (cchantep) [inline compiler/src/scala/quoted/runtime/impl/printers/SourceCode.scala:1240]: What would be the representation ?

File: compiler/src/scala/quoted/runtime/impl/printers/Extractors.scala [hunk 2]
```diff
@@ -240,6 +242,8 @@ object Extractors {
         this += "NoPrefix()"
       case MatchCase(pat, rhs) =>
         this += "MatchCase(" += pat += ", " += rhs += ")"
+      case tv: Types.TypeVar =>
```
Comment (cchantep) [inline compiler/src/scala/quoted/runtime/impl/printers/Extractors.scala:245]: Doesn't see how to test it, check that it fix the issue where it was seen

**Comment by cchantep** [comment]
That's quite a blocker, as it prevents from debuging, and even block `-Xcheck-macros` (and it uses `.show`)

**Comment by nicolasstucki** [comment]
We need to fix the code that leaks the `TypeVar` and not add patches if it appears somewhere it should not. See https://github.com/lampepfl/dotty/issues/13895#issuecomment-968869092, the fix is probably trivial but we need to be able to reproduce it.

**Comment by nicolasstucki** [comment]
We first need to do https://github.com/lampepfl/dotty/issues/13895#issuecomment-968869092 before we can fix this issue. Closing the PR for now.

**Pull request closed without merging**